### PR TITLE
AP_Notify: Add mah consumed to bat line in OLED display

### PR DESCRIPTION
**Pull request by Hwurzburg**
P_Jan29_230306
built and tested including field shifts due to range changes (but tested on 4.0.1 since master does not build without flash overflow on omnibusf4pro now...hope its fixed soon)

### CODE REVIEW

**Comment by tridge** [comment]
nice idea. Would it be better to display percentage remaining though? I think "73%" is clearer than a mAh value

**Comment by Hwurzburg** [comment]
changed as requested....board tested

File: libraries/AP_Notify/Display.cpp [hunk 1]
```diff
@@ -524,7 +524,9 @@ void Display::update_ekf(uint8_t r)
 void Display::update_battery(uint8_t r)
 {
     char msg [DISPLAY_MESSAGE_SIZE];
-    snprintf(msg, DISPLAY_MESSAGE_SIZE, "BAT1: %4.2fV", (double)AP::battery().voltage()) ;
+    AP_BattMonitor &battery = AP::battery();
+    uint8_t pct = battery.capacity_remaining_pct();
+    snprintf(msg, DISPLAY_MESSAGE_SIZE, "BAT:%4.2fV %2d%%left ", (double)AP::battery().voltage(), pct) ;
```
Comment (khancyr) [inline libraries/AP_Notify/Display.cpp]: to be pedantic, you could use battery.voltage() here

Comment (tridge) [reply libraries/AP_Notify/Display.cpp]: yes, should be battery.voltage()

Comment (Hwurzburg) [reply libraries/AP_Notify/Display.cpp]: do you want me to change? it was historical

Comment (Hwurzburg) [reply libraries/AP_Notify/Display.cpp]: > yes, should be battery.voltage()

it will not build with that

File: libraries/AP_Notify/Display.cpp [hunk 2]
```diff
@@ -524,7 +524,9 @@ void Display::update_ekf(uint8_t r)
 void Display::update_battery(uint8_t r)
 {
     char msg [DISPLAY_MESSAGE_SIZE];
-    snprintf(msg, DISPLAY_MESSAGE_SIZE, "BAT1: %4.2fV", (double)AP::battery().voltage()) ;
+    AP_BattMonitor &battery = AP::battery();
```
Comment (peterbarker) [inline libraries/AP_Notify/Display.cpp]: Can this be `const`?

File: libraries/AP_Notify/Display.cpp [hunk 3]
```diff
@@ -524,7 +524,9 @@ void Display::update_ekf(uint8_t r)
...
+    uint8_t pct = battery.capacity_remaining_pct();
```
Comment (peterbarker) [inline libraries/AP_Notify/Display.cpp]: Can be `const`

File: libraries/AP_Notify/Display.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_Notify/Display.cpp]: changed as requested

**Pull request merged**